### PR TITLE
Rewrite [dynamicxml] tests, and fix issue with attribute filters

### LINKED ISSUE #3089: Use fixtures for tests of dynamic badges
The dynamic badge tests depend on live data, and it's one more thing that can break: https://github.com/badges/shields/pull/3049#issuecomment-466599836

There's no benefit to having the dynamic badges depend on data that must be fetched live. It'd be simpler and more self-documenting to make 100% of the tests use fixtures and `intercept()`.

### PR DESCRIPTION
**Pull request by paulmelnikow**
This fixes queries with an attribute filter in the middle, such as `//book[@id='bk102']/title`.

Close #3089

### CODE REVIEW

File: services/dynamic/dynamic-xml.tester.js [hunk 1]
```diff
@@ -111,46 +133,18 @@ t.create('XML from url | invalid url')
...
-  .get('.json?url=https://xml-test/api.xml&query=/name')
   .intercept(nock =>
-    nock('https://xml-test')
-      .get('/api.xml')
-      .reply(200, function(uri, requestBody) {
-        headers = this.req.headers
-        return '<?xml version="1.0" encoding="utf-8" ?><name>dynamic xml</name>'
-      })
+    nock('https://example.test', {
+      reqheaders: { accept: 'application/xml, text/xml' },
```
Comment (paulmelnikow) [inline services/dynamic/dynamic-xml.tester.js]: This causes a "Mocks not net satisfied" assertion to be thrown if the Accept header doesn't match.

**Comment by calebcartwright** [comment]
Nice. Just to validate my understanding of this/#3089, using a live endpoint to retrieve XML has been making the tests problematic, and there's no real value in retrieving the XML from a live endpoint anyway (not like there's a static CircleCI, etc. service endpoint that live tests would be validating)

**Comment by paulmelnikow** [comment]
Yea, that's exactly it.

File: services/dynamic/dynamic-xml.tester.js [hunk 2]
```diff
@@ -25,76 +28,95 @@ t.create('No query specified')
...
   .expectBadge({
     label: 'custom badge',
-    message: Joi.string().regex(
-      /^Firefox( for Android)?,\sFirefox( for Android)?$/
-    ),
+    message:
+      "XML Developer's Guide, Midnight Rain, Maeve Ascendant, Oberon's Legacy, The Sundered Grail, Lover Birds, Splish Splash, Creepy Crawlies, Paradox Lost, Microsoft .NET: The Programming Bible, MSXML3: A Comprehensive Guide, Visual Studio 7: A Comprehensive Guide",
   })
 
-t.create('XML from url | caching with new query params')
```
Comment (paulmelnikow) [inline services/dynamic/dynamic-xml.tester.js]: I removed this test ^^ Frankly I have no idea what it is for. It looks like it was copied and pasted, or modified into oblivion.

**Pull request merged**